subprojects/core-api/src/main/java/org/gradle/api/file/SourceDirectorySet.java

```diff
@@ -130,4 +130,31 @@
      */
     @Incubating
     void setOutputDir(File outputDir);
+
+    /**
+     * Returns the directory to place source files generated by annotation processors.
+     *
+     * @return The generated sources directory for this set of sources.
+     * @since 5.1
+     */
+    @Incubating
+    File getAnnotationProcessorGeneratedSourcesDirectory();
```

Comment:
❌ These should not be here. Having `setOutputDir` here was a mistake that we still haven't fixed. A SourceDirectorySet is just what the name says - a set of source directories. It shouldn't know anything about output.